src/main/java/coupon/repository/CouponReader.java

```diff
@@ -0,0 +1,20 @@
+package coupon.repository;
+
+import coupon.domain.Coupon;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.UUID;
+
+@Component
+@Transactional(readOnly = true)
+@RequiredArgsConstructor
+public class CouponReader {
+    private final CouponRepository couponRepository;
+
+    public Coupon findById(final UUID uuid){
+        return couponRepository.findById(uuid)
+                .orElseThrow(()->new IllegalArgumentException(String.format("%s 에 해당하는 쿠폰이 없습니다.",uuid)));
+    }
```

Comment:
```suggestion
    public Coupon findById(final UUID uuid) {
        return couponRepository.findById(uuid)
                .orElseThrow(() -> new IllegalArgumentException(String.format("%s 에 해당하는 쿠폰이 없습니다.",uuid)));
    }
```
띄어쓰기 문법이 제대로 지켜지지 않은 곳이 있네요! 전체적으로 점검 한 번 해볼까요?

Reply:
ㅋㄱㅋㄱㅋㄱㅋㄱㄲ 죄송해요.
프로젝트에서도 자주 리뷰 달립니다. 🥲🥲